### PR TITLE
chore(tmc): add pagination support to /v1/stacks.

### PR DESCRIPTION
**Pull request by i4ki**
## What this PR does / why we need it:

Uses pagination when requesting resources from `v1/stacks/...` endpoints.

## Does this PR introduce a user-facing change?
```
no
```

### CODE REVIEW

File: cloud/cloud.go [hunk 1]
```diff
@@ -80,12 +83,21 @@ type (
...
+	pageSize         int64 = defaultPageSize
+	debugAPIRequests bool
+)
 
 func init() {
 	if d := os.Getenv("TMC_API_DEBUG"); d == "1" || d == "true" {
 		debugAPIRequests = true
 	}
+	if sizeStr := os.Getenv("TMC_API_PAGESIZE"); sizeStr != "" {
+		size, _ := strconv.Atoi64(sizeStr)
```
Comment (wmalik) [inline cloud/cloud.go:96]: Should we handle the error here and print a warning?

File: strconv/format.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import "strconv"
+
+// Itoa64 returns the string representation of the provided int64 using base 10.
+func Itoa64(i int64) string { return strconv.FormatInt(i, 10) }
+
+// Atoi64 returns the int64 represented in the given string.
+func Atoi64(a string) (int64, error) {
+	i, err := strconv.ParseInt(a, 10, 64)
+	return i, err
```
Comment (wmalik) [inline strconv/format.go:14]: nitpick: 
```suggestion
	return strconv.ParseInt(a, 10, 64)
```

**Review by wmalik — COMMENTED** [review]
Added some minor inline comments.

Regarding the changes in this PR, I see three kinds of changes:

* stack.ID type change (int -> int64)
* new helper method `(c *Client) URL(...)`
* The pagination implementation

It's not clear to me if the first two changes are related to the pagination implementation, it would be nice to mention that in the commit message if that's the case. Otherwise we should split the commits if possible.

**Comment by i4ki** [comment]
@wmalik I created a separated PR with the int64 changes: #1308 

About the `c.URL()` it was created as a URL builder supporting the query parameters for the pagination and other fields.

File: cloud/cloud.go (hunk 1 above)
Comment (i4ki) [reply cloud/cloud.go:96]: This is just for debugging purposes.

**Review by wmalik — APPROVED** [review]

**Pull request merged**